### ISSUE TITLE
Minimal OP_PUSHDATA in scripts

### ISSUE THREAD

**Issue by fanatid** [bitcoinjs/bitcoinjs-lib#480]
script.isScriptHashOutput has next bug:
```js
bjs = require('./src')
buf = new Buffer('a94c14c286a1af0947f58d1ad787385b1c2c4a976f9e7187', 'hex')
console.log(bjs.script.isScriptHashOutput(buf)) // true
```
bitcoind (decodescript):
```json
{
"asm" : "OP_HASH160 c286a1af0947f58d1ad787385b1c2c4a976f9e71 OP_EQUAL",
"type" : "nonstandard",
"p2sh" : "2ND6G91xAxiq3hWHUokremQJBuhyQKTVxWj"
}
```

function should be like something next:
```js
function isScriptHashOutput (buffer) {
  typeforce(types.Buffer, buffer)

  return buffer.length === 23 &&
    buffer[0] === OPS.OP_HASH160 &&
    buffer[1] === 0x14 &&
    buffer[22] === OPS.OP_EQUAL
}
```

small notes in [bitcoin code](https://github.com/bitcoin/bitcoin/blob/2b0567b00291205731cbb910fb3cf33359a046ee/src/script/standard.cpp#L61)

**Comment by dcousens** [comment]
Not technically a bug,  its just that the core client has become more strict about excessive `OP_PUSHDATA1`.

I might make `isScriptHashOutput` have a `nonStrict`/`strict` parameter to make sure the script adheres to standard network rules...

**Comment by dcousens** [comment]
Actually,  since https://github.com/bitcoin/bips/blob/master/bip-0062.mediawiki was soft-forked,  I guess it is a bug now.
Will release a patch version rather than minor.

**Comment by dcousens** [comment]
Nevermind,  this is still just a matter of `isStandard` as can be seen in [your reference](https://github.com/bitcoin/bitcoin/blob/2b0567b00291205731cbb910fb3cf33359a046ee/src/script/standard.cpp#L61).

Not technically a bug,  not sure what to do with this one.

**Comment by dcousens** [comment]
Please move the discussion [here](https://github.com/bitcoinjs/bitcoinjs-lib/pull/481).
**TL;DR** this is just a discussion of whether `bitcoinjs-lib` should enforce `IsStandard` rules on the `is*` script functions...
